README.md

```diff
@@ -11,78 +11,83 @@ related-issues: (will contain links to implementation PRs)
   - First one has to find the definition of it in all-packages.nix to see what file it refers to
-    - Especially on GitHub this is even more problematic, as the `all-packages.nix` file is [too big to be displayed by GitHub](https://github.com/NixOS/nixpkgs/blob/nixos-22.05/pkgs/top-level/all-packages.nix)
+    - On GitHub this is even more problematic, as the `all-packages.nix` file is [too big to be displayed by GitHub](https://github.com/NixOS/nixpkgs/blob/nixos-22.05/pkgs/top-level/all-packages.nix)
   - Then go to that file's definition, which takes quite some time for navigation (unless you have a plugin that can jump to it directly)
+    - It also slows down or even deadlocks editors due to the file size
+  - `nix edit -f . package-attr` works, though that's not yet stable (it relies on the `nix-command` feature being enabled) and doesn't work with packages that don't set `meta.position` correctly).
 - `all-packages.nix` frequently causes merge conflicts. It's a point of contention for all new packages
 
 # Detailed design
 [design]: #detailed-design
 
-All attributes at the root of nixpkgs (`pkgs.<name>`) whose definition is of the form
-
-```nix
-{
-  <name> = pkgs.callPackage ../some/dir { };
-}
-```
-
-will be become eligible to be transformed as follows:
-- Move `pkgs/some/dir` to `pkgs/auto/<name>`
-- Fix any references to that folder (e.g. in update scripts) 
-- Remove the original definition
-
-These attributes will newly be defined by listing all directories in `pkgs/auto` using `builtins.readDir` and calling `pkgs.callPackage` on all of them, which then gets added to the `pkgs` scope.
-
-Attributes whose definition aren't of the above form won't be changed, so e.g. the following definition in `all-packages.nix` won't be changed:
-```nix
-{
-  syncplay-nogui = syncplay.override { enableGUI = false; };
-}
-```
-
-However, if such definitions can be refactored into the above form they will become eligible for the transformation.
+This RFC establishes the convention of `pkgs/unit/${substring 0 4 name}/${name}` "unit" directories for the definitions of the Nix packages `pkgs.${name}` in nixpkgs.
+The `pkg-fun.nix` files in all unit directories are automatically discovered, called using `pkgs.callPackage` and added to the `pkgs` set.
 
-## Backwards compatibility symlinks
-[symlinks]: #backwards-compatibility-symlinks
+These requirements will be checked using CI:
+1. <a id="req-1"/> The `pkgs/unit` directory must only contain unit directories, and only in subdirectories of the form `${substring 0 4 name}/${name}`.
+2. <a id="req-2"/> Files outside a unit directory must not reference files inside that unit directory, and the other way around.
+3. <a id="req-3"/> Each unit directory must have a `pkg-fun.nix` file such that `pkgs.callPackage ./pkg-fun.nix {}` evaluates to a derivation.
+4. <a id="req-4"/> Packages defined by unit directories must not be defined or overridden anywhere else, such as in `pkgs/top-level/all-packages.nix`.
```

Comment:
I also codified these requirements that should be checked by CI

1. So that people don't pollute the `pkgs/unit` directory with wrong shards
2. Makes the unit self-contained which is a good thing to have in general, and we get this check essentially for free with the tooling for the transition. If this becomes a point of contention we could also remove this requirement though.
3. So we can always `callPackage`
4. So that we don't have one person adding a file in `all-packages.nix` and another in `pkgs/unit`, and one of them being ignored without any indication.